src/main/java/org/spongepowered/common/entity/player/SpongeUserData.java

```diff
@@ -118,24 +113,53 @@
     private boolean isInvisible;
     private boolean vanishIgnoresCollision;
     private boolean vanishPreventsTargeting;
+    private final GameProfile profile;
 
-    private @Nullable SubjectReference subjectReference;
     private @Nullable SpongeUserInventory inventory; // lazy load when accessing inventory
     private @Nullable PlayerEnderChestContainer enderChest; // lazy load when accessing inventory
-    private @Nullable CompoundTag compound;
+    private CompoundTag compound;
     private boolean isConstructing;
 
-    public SpongeUser(final GameProfile profile) {
+    public static SpongeUserData create(final GameProfile profile) throws IOException {
+        final ServerLevel world = SpongeCommon.server().overworld();
+        if (world == null) {
+            SpongeCommon.logger().warn("Overworld not initialized, cannot create users!");
+            throw new IllegalStateException("Overworld not initialized, cannot create users!");
+        }
+
+        final LevelStorageSource.LevelStorageAccess storageSource = ((MinecraftServerAccessor) Sponge.server()).accessor$storageSource();
+        final File file = storageSource.getLevelPath(LevelResource.PLAYER_DATA_DIR).resolve(profile.getId().toString() + ".dat").toFile();
```

Comment:
Thought about migrating this to `Path` (java.nio) ?

Reply:
I'm not doing that here. This is mirroring what Vanilla does and I don't really want to break from that, the comparison is easier then.

Reply:
Fair enough